src/main/java/org/jabref/preferences/PreferencesService.java

```diff
@@ -35,4 +35,8 @@
 
     public void updateEntryEditorTabList();
 
+    List<String> getStringList(String key);
```

Comment:
Please no generic methods here, create specific ones for the cases you need, e.g. get CustomExporters or getExporterormat...  Or you create CustomExporter preferences which has all the needed information inside.

Reply:
The Swing version has a class CustomExportList which holds a list of custom exporters and associated data.  I was going to move some of the code here into the ExporterViewModel and other code into the ExporterCustomizationDialogViewModel.  Do you recommend keeping the CustomExportList as it is?

Reply:
The PreferencesService should only contain usecase-specific preference methods and the caller shouldn't care about how these information are stored in the preferences. So, here, the method would look like `List<Exporter> getCustomExportFormats()` and then in the implementation there would be calls like `preferences.getStringList(JabRefPreferences.CUSTOM_EXPORT_FORMAT + i))`.

Thus, most of the code in `CustomExportList` should go into `JabrefPreferences`.

Reply:
Shouldn't the methods in `preferencesService` match those invoked in `ExportCustomizationDialogViewModel` as `exporters.add(preferences.X)` where X is the method, rather than using `preferences.getStringList`?

Reply:
Sorry, I don't really understand what you mean. The PreferencesService should provide an interface to the preferences that hide the exact semantics of how the object is serialized. So abstractly, `X getX()` and `putX(X newValue)`, where `X` is the (model) object. The implementation in `JabRefPreferences` than has to worry about (de)serialization.

Reply:
I'm not a big fan of writing tests for the sake of it. My strategy is usually to program everything and when I discover a problem in the logic during testing, I'll write a test for it. Hence, if the dialog is simple enough that you don't make any errors, then you don't need to write tests ;-).
An exception is when there is really complex logic that is definitely worth testing.